client/src/components/EditShowForm.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from "react"
+import ErrorList from "./ErrorList"
+import translateServerErrors from "../services/translateServerErrors"
+
+const EditShowForm = ({ showId, userId, handleEditShow }) => {
+  const [editShow, setEditShow] = useState({
+    name: "",
+    description: "",
+  })
+
+  const [errors, setErrors] = useState([])
+
+  const handleInputChange = (event) => {
+    setEditShow({
+      ...editShow,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+    
+  const handleSubmit = async (event) => {
+    event.preventDefault()
+    const editedShow = editShow
+    editedShow.userId = userId
+
+    try {
+      const response = await fetch(`/api/v1/shows/${showId}`, {
+        method: "PATCH",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(editedShow)
+      })
+      if(!response.ok) {
+        if(response.status === 422) {
+          const body = await response.json()
+          const newErrors = translateServerErrors(body.errors)
+          return setErrors(newErrors)
+        } else {
+          const errorMessage = `${response.status}: (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw(error)
+        }
+      } else {
+        setErrors([])
+        clearForm()
+        const editedShow = await response.json()
```

Comment:
this `await.response.json` will be called regardless of if the response is okay or not (its also called on line 35). as such, I would move it just above line 33 so that we make the code a bit DRYer